x-pack/filebeat/input/httpjson/requester.go

```diff
@@ -229,15 +230,23 @@ func (r *requester) processEventArray(publisher stateless.Publisher, events []in
 	return last, nil
 }
 
-func (r *requester) splitEvent(event map[string]interface{}) []map[string]interface{} {
+func splitEvent(splitKey string, event map[string]interface{}) []map[string]interface{} {
 	m := common.MapStr(event)
 
-	hasSplitKey, _ := m.HasKey(r.splitEventsBy)
-	if r.splitEventsBy == "" || !hasSplitKey {
+	// NOTE: this notation is only used internally, not meant to be documented
+	// and will be removed in the next release
+	keys := strings.SplitN(splitKey, "..", 2)
```

Comment:
I know this is unlikely but this means what you split on can't have ".." in it.  Could we maybe turn splitKey into an Array (here and in config).  Then you could have any value, and it could be arbitrarily deep.

Reply:
That would be a breaking change, so I'm taking back the suggestion.